src/stories/components/JsPsych/JsPsych.js

```diff
@@ -0,0 +1,240 @@
+import example2 from '../../../images/example2.png';
+
+/**
+ * Component to load jsPsych's "Self Image Experiment".
+ *
+ * @component
+ * @return {object} (
+ *   <React.Fragment />
+ * )
+ */
+export const JsPsych = () => {
+  const { experimentId, participantId } = useParams(); // Parse URL params
+  const [stimuliUrls, setStimuliUrls] = useState([]);
+  const [ready, setReady] = useState(false);
+
+  useEffect(() => {
+    // Load jsPsych library core script and styling & plugins used in the experiment
+    const jsPsychLibrary = document.createElement('script');
+    const jsPsychStyleSheet = document.createElement('link');
+    const htmlKeyboardResponsePlugin = document.createElement('script');
+    const imageKeyboardResponsePlugin = document.createElement('script');
+    jsPsychLibrary.src = '../../../external_libraries/jspsych-6.3.0/jspsych.js';
+    jsPsychStyleSheet.href = '../../../external_libraries/css/jspsych.css';
+    jsPsychStyleSheet.rel = 'stylesheet';
+    jsPsychStyleSheet.type = 'text/css';
+    htmlKeyboardResponsePlugin.src = '../../../external_libraries/jspsych-6.3.0/plugins/jspsych-html-keyboard-response.js';
+    imageKeyboardResponsePlugin.src = '../../../external_libraries/jspsych-6.3.0/plugins/jspsych-image-keyboard-response.js';
+    jsPsychLibrary.async = true;
+    jsPsychStyleSheet.async = true;
+    htmlKeyboardResponsePlugin.async = true;
+    imageKeyboardResponsePlugin.async = true;
+    document.body.appendChild(jsPsychLibrary);
+    document.body.appendChild(jsPsychStyleSheet);
+    document.body.appendChild(htmlKeyboardResponsePlugin);
+    document.body.appendChild(imageKeyboardResponsePlugin);
+
+    // Load participantId, experimentId, and 400 stimuli urls
+    observeStimuliCompletion(participantId, experimentId, setStimuliUrls, errorLoadingJsPsych);
+
+    return () => {
+      document.body.removeChild(jsPsychLibrary);
+      document.body.removeChild(jsPsychStyleSheet);
+      document.body.removeChild(htmlKeyboardResponsePlugin);
+      document.body.removeChild(imageKeyboardResponsePlugin);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (stimuliUrls.length > 0) {
+      setReady(true);
+    }
+  }, [stimuliUrls]);
+
+  // Error handler that prompts the participant to re-click experiment.
+  const errorLoadingJsPsych = (errorCode) => {
+    window.alert('Something went wrong. Please click on experiment again.' +
+      ' Error code: ' + errorCode);
+  };
+
+  // Have to wait for the plugin scripts to be loaded asynchronously before
+  // running the experiment; otherwise it crashes.
+  setTimeout(() => {
+    // Implementation of previous team timeline & trial logic -------------------------- START
+    /* create timeline */
+    const timeline = [];
+
+    /* number of trials */
+    // TODO: Adjust NUMBER_OF_TRIALS back to 199
+    // NOTE: Adjust line below to shorten the number of trials. 199 will go through all 200 iterations.
+    const NUMBER_OF_TRIALS = 5;
```

Comment:
Note: we will have to adjust this number back to 199 before launching / going live. I've shortened it to 5 (or a total of 6 trials) for ease of testing at the moment.